x/sequencer/keeper/slashing.go

```diff
@@ -31,20 +31,34 @@ func (k Keeper) Slashing(ctx sdk.Context, seqAddr string) error {
 		k.Logger(ctx).Error("sequencer has no tokens to slash", "sequencer", seq.SequencerAddress)
 	}
 	seq.Tokens = sdk.Coins{}
-
 	oldStatus := seq.Status
-	wasProposer := seq.Proposer
-	// in case we are slashing an unbonding sequencer, we need to remove it from the unbonding queue
+
+	// remove from queue if unbonding
 	if oldStatus == types.Unbonding {
 		k.removeUnbondingSequencer(ctx, seq)
 	}
+	// remove from notice period queue if needed
+	if seq.IsNoticePeriodInProgress() {
+		k.removeNoticePeriodSequencer(ctx, seq)
+	}
+	// if the slashed sequencer is the proposer, remove it
+	// the caller should rotate the proposer
+	if k.isProposer(ctx, seq.RollappId, seqAddr) {
+		k.removeProposer(ctx, seq.RollappId)
+	}
+
+	// if we slash the next proposer, we're in the middle of rotation
+	// instead of removing the next proposer, we set it to empty
+	if k.isNextProposer(ctx, seq.RollappId, seqAddr) {
+		k.setNextProposer(ctx, seq.RollappId, "")
+	}
```

Comment:
Slashing is now very coupled to the proposer rotation
What is supposed to happen in case of partial slash?

Reply:
i'll move the code to unbond handler
than the slashing logic should decide whether unbond required